packages/blade/src/components/Input/BaseInput/__tests__/BaseInput.web.test.tsx

```diff
@@ -0,0 +1,157 @@
+
+    const input = getByLabelText(label);
+    expect(input).toBeDisabled();
+  });
+
+  it('should handle onChange', async () => {
+    const label = 'Enter name';
+    const onChange = jest.fn();
+    const user = userEvent.setup();
+    const userName = 'Divyanshu';
+
+    const { getByLabelText } = renderWithTheme(
+      <BaseInput label={label} id="name" name="name" onChange={onChange} />,
+    );
+
+    const input = getByLabelText(label);
+    await user.type(input, userName);
+
+    // should be called for each keystroke
+    expect(onChange).toHaveBeenCalledTimes(userName.length);
+    expect(onChange).toHaveBeenLastCalledWith({ name: 'name', value: userName });
+  });
+
+  it('should handle onBlur', async () => {
+    const user = userEvent.setup();
+    const label = 'Enter name';
+    const userName = 'Divyanshu';
+    const onBlur = jest.fn();
+
+    renderWithTheme(
+      <BaseInput
+        label={label}
+        id="name"
+        name="name"
+        defaultValue={userName}
+        // eslint-disable-next-line jsx-a11y/no-autofocus
+        autoFocus
+        onBlur={onBlur}
+      />,
+    );
+
+    // shifts user focus and therefore blurs the focussed input
+    await user.tab();
+    expect(onBlur).toHaveBeenCalledTimes(1);
+    expect(onBlur).toHaveBeenCalledWith({ name: 'name', value: userName });
+  });
+
+  it('should set value as an uncontrolled input', async () => {
+    const user = userEvent.setup();
+    const label = 'Enter name';
+    const valueInitial = 'Divyanshu';
+    const valueFinal = 'Divyanshu Maithani';
+
+    const { getByLabelText } = renderWithTheme(
+      <BaseInput label={label} id="name" defaultValue={valueInitial} />,
+    );
+
+    const input = getByLabelText(label);
+    expect(input).toHaveValue(valueInitial);
+
+    await user.type(input, ' Maithani');
+    expect(input).toHaveValue(valueFinal);
+  });
+
+  it('should set value as a controlled input', () => {
+    const label = 'Enter name';
+    const valueInitial = 'Divyanshu';
+    const valueFinal = 'Divyanshu Maithani';
+
+    const { getByLabelText, rerender } = renderWithTheme(
+      <BaseInput label={label} id="name" value={valueInitial} />,
+    );
+
+    const input = getByLabelText(label);
+    expect(input).toHaveValue(valueInitial);
+
+    // simulate a controlled input by updating value prop
+    rerender(
+      <ThemeWrapper>
+        <BaseInput label={label} id="name" value={valueFinal} />
+      </ThemeWrapper>,
+    );
+    expect(getByLabelText(label)).toHaveValue(valueFinal);
+  });
+
+  it('should pass a11y', async () => {
+    const { getByRole } = renderWithTheme(
+      <BaseInput
+        label="Enter name"
+        id="name"
+        isRequired
+        helpText="First name and last name"
+        validationState="none"
+      />,
+    );
+
+    const input = getByRole('textbox');
+    await assertAccessible(input);
+  });
```

Comment:
we shall also add tests for 
* error text since the input becomes invalid
* same for success text
and check that the input is marked accessible along with necessary attributes which associates errorText and successText with the input

Reply:
Done in https://github.com/razorpay/blade/pull/695/commits/1b30d066d0edf03c62601f0c8a2dbb64fe5b7b31